### PR TITLE
add configurable traits to api docs

### PR DESCRIPTION
**Pull request by minrk**
Any `config=True` trait added to these classes will get automatically included in the docs.

### CODE REVIEW

File: docs/sphinxext/autodoc_traits.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    def can_document_member(cls, member, membername, isattr, parent):
+        return isinstance(member, TraitType)
+
+    def format_name(self):
+        return 'config c.' + super().format_name()
+
+    def add_directive_header(self, sig):
+        sig = ' = {}(default={!r})'.format(
+            self.object.__class__.__name__,
+            self.object.default(),
```
Comment (willingc) [inline docs/sphinxext/autodoc_traits.py]: When trying to build (`make html`), I'm getting an error:
```
  File "/Users/carol/code/jhub/jupyterhub/docs/sphinxext/autodoc_traits.py", line 42, in add_directive_header
    self.object.default(),
AttributeError: 'Set' object has no attribute 'default'
```

Comment (minrk) [inline docs/sphinxext/autodoc_traits.py]: I must have been relying on something in unreleased traitlets. Will fix.

Comment (willingc) [inline docs/sphinxext/autodoc_traits.py]: Latest changes to traitlets fix this.

File: docs/source/api/index.rst [hunk 2]
```diff
@@ -24,6 +24,7 @@ JupyterHub API Reference:
 
 .. toctree::
 
+    app
```
Comment (willingc) [inline docs/source/api/index.rst]: Causing a build failure in docs since `app.rst` does not exist in `docs/source/api` directory.

**Comment by willingc** [comment]
Thanks @minrk. I gave Travis a nudge and all tests pass now.

**Pull request merged**